Decred Wallet/Features/Send/SendViewController.swift

```diff
@@ -503,37 +503,43 @@ extension SendViewController {
     }
     
     func checkAddressFromQrCode(textScannedFromQRCode: String?) {
-        guard var capturedText = textScannedFromQRCode else {
+        
+        guard let capturedText = textScannedFromQRCode else {
             self.destinationAddressTextField.text = ""
             return
         }
+        let URI: DecredAddressURI = DecredAddressURI(uriString: capturedText)
         
-        if capturedText.starts(with: "decred:") {
-            capturedText = capturedText.replacingOccurrences(of: "decred:", with: "")
+        guard let address = URI.address else {
+            return
         }
         
-        if capturedText.count < 25 {
+        if address.count < 25 {
             self.invalidAddressFromQrCode(errorMessage: LocalizedStrings.walletAddressShort)
```

Comment:
The related dcrandroid issue (https://github.com/decred/dcrandroid/issues/407) also requests that the app no longer show too long or too short address error messages. It also suggests displaying the scanned text to the user to help the user determine why the scanned text is invalid.
```
error message "address is too long" should not be shown, instead it would be, not a valid address.
Can show the user the contents of what they scanned for debugging purposes.
```

Reply:
resolved in `c32ac54`